external/fv3fit/fv3fit/pytorch/optimizer.py

```diff
@@ -17,3 +20,27 @@ def instance(self, params):
         kwargs = dict(params=params, **self.kwargs)
 
         return cls(**kwargs)
+
+
+class Scheduler(Protocol):
+    def step(self):
+        ...
+
+
+class NullScheduler:
+    def step(self):
+        pass
+
+
+@dataclasses.dataclass
+class SchedulerConfig:
+    name: Optional[str] = None
+    kwargs: Mapping[str, Any] = dataclasses.field(default_factory=dict)
+
+    def instance(self, optimizer) -> Scheduler:
+        if self.name is None:
+            return NullScheduler()
```

Comment:
[nit] this is technically not of type `Scheduler`, should it be `class NullScheduler(Scheduler)`?

Reply:
Scheduler is a protocol type, meaning you don't have to subclass it to be an instance of it. `isinstance(null_scheduler, Scheduler)` would be True for `null_scheduler: NullScheduler`.